### PR TITLE
Fix matrix factorization trainer's doc based on user feedback

### LINKED ISSUE #3169: KeyType in example is invalid
[Enter feedback here]
In 0.11 KeyType only takes 1 parameter -  'ulong count' 
class MatrixElement
       {
           // Matrix column index starts from firstColumnIndex and is at most firstColumnIndex+n-1.
           // Contieuous=true means that all values from firstColumnIndex to firstColumnIndex+n-1 are allowed keys.
           // [KeyType(Contiguous = true, Count = n, Min = firstColumnIndex)]
           // public uint MatrixColumnIndex;
           // Matrix row index starts from firstRowIndex and is at most firstRowIndex+m-1.
           // Contieuous=true means that all values from firstRowIndex to firstRowIndex+m-1 are allowed keys.
           [KeyType(Contiguous = true, Count = m, Min = firstRowIndex)]
           public uint MatrixRowIndex;
           // The rating at the MatrixColumnIndex-th column and the MatrixRowIndex-th row.
           public float Value;
       }
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 335e2f1c-de96-9963-8edc-c45bc5d31105
* Version Independent ID: 571d5df2-bce7-9fdf-c123-5c932d7935ce
* Content: [MatrixFactorizationTrainer Class (Microsoft.ML.Trainers)](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers.matrixfactorizationtrainer?view=ml-dotnet)
* Content Source: [dotnet/xml/Microsoft.ML.Trainers/MatrixFactorizationTrainer.xml](https://github.com/dotnet/ml-api-docs/blob/live/dotnet/xml/Microsoft.ML.Trainers/MatrixFactorizationTrainer.xml)
* Product: **dotnet-ml-api**
* GitHub Login: @sfilipi
* Microsoft Alias: **johalex**

### PR DESCRIPTION
**Pull request by wschin**
This PR fixes #3169 by
- updating key-typed variable's description in matrix factorization trainer's doc, and
- adding one more reference for alg used.

### CODE REVIEW

**Comment by artidoro** [comment]
>     /// }

It might be good to avoid this kind of issues in the future to include the code sample that you are giving here in the sample file.
For maintainability purposes if we check that the samples compile it would be better. 

---
Refers to: src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs:50 in 95f09fd. [](commit_id = 95f09fde4bdec993d40640f72cacf1c7fdda0e03, deletion_comment = False)

**Comment by sfilipi** [comment]
>     /// }

+1

---
In reply to: [479781691](https://github.com/dotnet/machinelearning/pull/3170#issuecomment-479781691) [](ancestors = 479781691)

---
Refers to: src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs:50 in 95f09fd. [](commit_id = 95f09fde4bdec993d40640f72cacf1c7fdda0e03, deletion_comment = False)

File: src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs [hunk 1]
```diff
@@ -31,24 +31,19 @@ namespace Microsoft.ML.Trainers
...
     /// class MatrixElement
     /// {
-    ///     // Matrix column index starts from firstColumnIndex and is at most firstColumnIndex+n-1.
-    ///     // Contieuous=true means that all values from firstColumnIndex to firstColumnIndex+n-1 are allowed keys.
-    ///     // [KeyType(Contiguous = true, Count = n, Min = firstColumnIndex)]
-    ///     // public uint MatrixColumnIndex;
-    ///     // Matrix row index starts from firstRowIndex and is at most firstRowIndex+m-1.
-    ///     // Contieuous=true means that all values from firstRowIndex to firstRowIndex+m-1 are allowed keys.
-    ///     [KeyType(Contiguous = true, Count = m, Min = firstRowIndex)]
+    ///     // Matrix column index starts from 0 and is at most n-1.
```
Comment (sfilipi) [inline src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs]: >// Matrix column index starts from 0 and is at most n-1. [](start = 12, length = 56)

tangential, what do you think about #3072

File: src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs [hunk 2]
```diff
@@ -65,7 +60,7 @@ namespace Microsoft.ML.Trainers
     /// <i>R</i> is approximated by the product of <i>P</i>'s transpose and <i>Q</i>. This trainer implements
     /// <a href='https://www.csie.ntu.edu.tw/~cjlin/papers/libmf/mf_adaptive_pakdd.pdf'>a stochastic gradient method</a> for finding <i>P</i>
     /// and <i>Q</i> via minimizing the distance between<i> R</i> and the product of <i>P</i>'s transpose and Q.</para>.
-    /// <para>For users interested in the mathematical details, please see the references below.</para>
+    /// <para>The underlying library used in ML.NET matrix factorization can be found on <a href='https://github.com/cjlin1/libmf'>a Github repository</a>. For users interested in the mathematical details, please see the references below.</para>
```
Comment (sfilipi) [inline src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs]: >on <a href='https://github.com/cjlin1/libmf'>a Github repository</a> [](start = 86, length = 68)

on the <a href='[https://github.com/cjlin1/libmf](https://github.com/cjlin1/libmf)'>libmf Github repository</a>

**Pull request merged**